### DISCUSSION TITLE
Dagster versions with gRPC

### DISCUSSION THREAD

**Discussion by asharov** [dagster-io/dagster#3023] [Q&A] [answered]
Hi all,

I have a setup with pipelines running on a gRPC server, and I'm wondering how to go about updating dagster. Can I update the dagster versions for dagit and for the pipeline server independently or do the versions need to match? And if they need to match, by how much? Exact same version, same minor version, something else?

Thanks

**Comment by alangenfeld** [comment, marked as answer]
At this point in time with the gRPC servers being an opt-in feature we only guarantee everything will work if you keep the versions synced exactly. As we work towards making the gRPC servers the default, the plan is to make the versioning compatibility as flexible as we can. Will report back with more details when they are ready.

**Reply by asharov** [reply]
Thanks for the response! And glad to hear this will improve, I can manage for now with updating everything together.
